api/database/migrations/2022_10_20_170113_migrating_pool_candidate_data.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Support\Facades\DB;
+use App\Models\PoolCandidate;
+use App\Models\User;
+
+class MigratingPoolCandidateData extends Migration
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+
+    public function up()
+    {
+        // update table users with data from table pool_candidates
+        DB::statement('UPDATE users
+                    SET is_woman = p.is_woman,
+                        has_disability = p.has_disability,
+                        is_indigenous = p.is_indigenous,
+                        is_visible_minority = p.is_visible_minority,
+                        has_diploma = p.has_diploma,
+                        language_ability = p.language_ability,
+                        location_preferences = p.location_preferences,
+                        accepted_operational_requirements = p.accepted_operational_requirements,
+                        expected_salary = p.expected_salary
+                    FROM pool_candidates p
+                    WHERE users.id = p.user_id');
+
+        // insert data into cmo_users and classifications_user using PoolCandidate and User models with query builder
+        $allPoolCandidates = PoolCandidate::all();
+        foreach($allPoolCandidates as $candidate){
+            $userId = $candidate->user_id;
+
+            // build arrays of Assets and Classifications existing on PoolCandidate model and attached User
+            $classificationsCandidate = $candidate->expectedClassifications()->pluck('classifications.id')->toArray();
+            $assetsCandidate = $candidate->cmoAssets()->pluck('cmo_assets.id')->toArray();
+            $classificationsUser = User::where('id', $userId)->first()->expectedClassifications()->pluck('classifications.id')->toArray();
+            $assetsUser = User::where('id', $userId)->first()->cmoAssets()->pluck('cmo_assets.id')->toArray();
+
+            // merge into a combined Assets array as well as Classifications array, then remove duplicates
+            $combinedClassifications = array_merge($classificationsCandidate, $classificationsUser);
+            $combinedClassificationsDedupe = array_unique($combinedClassifications);
+            $combinedAssets = array_merge($assetsCandidate, $assetsUser);
+            $combinedAssetsDedupe = array_unique($combinedAssets);
+
+            // sync the User with the de-duped arrays
+            User::where('id', $userId)->first()->expectedClassifications()->sync($combinedClassificationsDedupe);
+            User::where('id', $userId)->first()->cmoAssets()->sync($combinedAssetsDedupe);
+        }
+    }
```

Comment:
Can we rewrite this to avoid using the PHP models? 

Basically, our PHP code might change. eg Maybe we remove the expectedClassifications relationship from PoolCandidate. But we won't want this migration to break, because it still needs to run every time we spin up a new dev environment. 

The database will always have the same shape every time this migration runs, since migrations run in deterministic order, so we can ensure it doesn't break by using [DB queries](https://laravel.com/docs/5.0/database#running-queries) instead of PHP Eloquent models.

For example, this line
```
$classificationsCandidate = $candidate->expectedClassifications()->pluck('classifications.id')->toArray();
```
could be replaced with 
```
$classificationsCandidate = DB::select("select candidate_id from classification_pool_candidate where pool_candidate_id = ?", $classificationId);
```